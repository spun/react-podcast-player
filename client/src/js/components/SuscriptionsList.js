import React from "react";
import PropTypes from "prop-types";
import { Link, NavLink } from "react-router-dom";

const SuscriptionsList = ({ list }) => {
	return (
		<div>
			{/* Check if the list is empty*/}
			{list.length == 0
				? /* If the list is empty */
					<div className="center">
						<p className="text-center">
							Your susctiption list is empty. <br />
							<br />
							<Link to={"/add"}>
								<span className="glyphicon glyphicon-plus" /> Add
								a new suscription to start listening.
							</Link>
						</p>
					</div>
				: /* If the list have suscriptions to show */
					<div>
						<ul className="nav suscriptions-list">
							{list.map((suscription, i) =>
								<li key={i}>
									<NavLink
										to={`/podcasts/${suscription.feed_id}`}
										activeStyle={{
											fontWeight: "bold",
											color: "red"
										}}>
										<p className="title">
											{suscription.title}
										</p>

										{suscription.unlistened > 0
											? <p className="author text-info">
													{suscription.unlistened}{" "}
													unlistened audios
												</p>
											: <p className="author text-muted">
													All listened
												</p>}
									</NavLink>
								</li>
							)}
						</ul>
					</div>}
		</div>
	);
};

SuscriptionsList.propTypes = {
	list: PropTypes.arrayOf(
		PropTypes.shape({
			suscription_id: PropTypes.number.isRequired,
			title: PropTypes.string.isRequired
		})
	)
};

export default SuscriptionsList;
