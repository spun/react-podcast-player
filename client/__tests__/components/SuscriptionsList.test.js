import React from "react";
import { MemoryRouter } from "react-router-dom";
import renderer from "react-test-renderer";
// Component under test
import SuscriptionsList from "../../src/js/components/SuscriptionsList";

test("SuscriptionsList creation with items", () => {
	const listIn = [
		{
			suscription_id: 1,
			title: "Suscription 1"
		},
		{
			suscription_id: 2,
			title: "Suscription 2"
		}
	];

	const component = renderer.create(
		<MemoryRouter>
			<SuscriptionsList list={listIn} />
		</MemoryRouter>
	);
	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});
