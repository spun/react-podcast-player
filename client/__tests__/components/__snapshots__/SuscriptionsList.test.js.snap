// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SuscriptionsList creation with items 1`] = `
<div>
  <div>
    <ul
      className="nav suscriptions-list"
    >
      <li>
        <a
          className={undefined}
          href="/podcasts/undefined"
          onClick={[Function]}
          style={undefined}
        >
          <p
            className="title"
          >
            Suscription 1
          </p>
          <p
            className="author text-muted"
          >
            All listened
          </p>
        </a>
      </li>
      <li>
        <a
          className={undefined}
          href="/podcasts/undefined"
          onClick={[Function]}
          style={undefined}
        >
          <p
            className="title"
          >
            Suscription 2
          </p>
          <p
            className="author text-muted"
          >
            All listened
          </p>
        </a>
      </li>
    </ul>
  </div>
</div>
`;
